libsolutil/StringUtils.h

```diff
@@ -50,6 +50,13 @@ std::string quotedAlternativesList(std::vector<std::string> const& suggestions);
 /// If @a _startSuffix == @a _endSuffix, the empty string is returned.
 std::string suffixedVariableNameList(std::string const& _baseName, size_t _startSuffix, size_t _endSuffix);
 
+/// Decodes a URI with respect to %XX notation.
+/// No URI-validity verification is performed but simply the URI decoded into non-escaping characters.
+std::string decodeURI(std::string const& _uri);
+
+/// Encodes a string into a URI conform notation.
+std::string encodeURI(std::string const& _uri);
```

Comment:
Given that these don't have to be URIs (it works for arbitrary strings), I'd call it something like `urlEncode()` and `urlDecode()` (or even `urlencode()` and `urldecode()`). Also because I always associate the term "urlencoding" specifically with this kind of encoding while "encoding an URI" could mean any encoding.I would probably look for the former I wanted to find it.